### DISCUSSION TITLE
Settings compiler flags with fortls?

### DISCUSSION THREAD

**Discussion by kshar-75** [fortran-lang/fortls#193] [Q&A] [answered]
Hey, the first thing I've noticed with my first "Hello World!" program in fortran is that for some reason the editor places a `[-Wtabs]` warning on the print statement. This is an issue because this type of syntax is extremely common among fortran programmers. How do I modify the fortls settings so that I can pick compiler flags of my own?

**Comment by gnikit** [comment, marked as answer]
That is a `gfortran` warning coming from the linter (I assume you are using VS Code as you editor) not `fortls` i.e. this repo. Tabs in Fortran are not considered part of the standard, use spaces instead or disable the warning by `-Wno-tabs`

**Comment by endingly** [comment]
You can use cmake. Fortls should only provide autocomplete.
```cmake
file(GLOB_RECURSE SOURCE_FILES *.f90)
message("Source files: ${SOURCE_FILES}")
add_executable(${PROJECT_NAME} ${SOURCE_FILES})
target_compile_options(${PROJECT_NAME} 
    PRIVATE
        -Wno-line-truncation
        -Wno-unused-variable
        -Wno-integer-division
        -Wno-conversion
        -Wno-tabs
        -ffree-line-length-none
)

**Reply by gnikit** [reply]
fortls does not "only provide autocomplete", it offers a lot more than just completions. For a complete list please advise the docs: https://fortls.fortran-lang.org/features.html
